instrumentation-core/spring/starters/spring-starter/spring-starter.gradle

```
@@ -11,7 +11,9 @@ sourceCompatibility = '8'
 
 dependencies { 
   api group: "org.springframework.boot", name: "spring-boot-starter", version: versions.springboot
+  api group: 'org.springframework.boot', name: 'spring-boot-starter-aop', version: versions.springboot
   api project(':instrumentation-core:spring:spring-boot-autoconfigure')
+  api deps.opentelemetryApiAutoAnnotations
   api deps.opentelemetryApi
   api deps.opentelemetryLogging
   api deps.opentelemetrySdk
```

based on discussion in #846, does this make sense?
```suggestion
  api deps.opentelemetryApi
  compileOnly deps.opentelemetryApiAutoAnnotations
  compileOnly deps.opentelemetrySdk
```

---

For the starter I'd like to provide all the opentelemetry dependencies required to instrument a spring application. Users can create their own custom starters and supply their own implementation of the opentelemetry-sdk and opentelemetryApiAutoAnnotations using opentelemetry-spring-boot-autoconfigure project. Here's that gradle file:
 - https://github.com/mabdinur/opentelemetry-java-instrumentation/blob/add_with_span_spring_aop/instrumentation-core/spring/spring-boot-autoconfigure/spring-boot-autoconfigure.gradle. 

Here's a pull request that documents this behavior: 
https://github.com/mabdinur/opentelemetry-java-instrumentation/pull/14/files

---

What would make sense is removing the opentelemetrySdk dependency since this is provided by the default logger (opentelemetryLogging). I'm not sure if this change would reduce readability. 

```suggestion
  api deps.opentelemetryApiAutoAnnotations
  api deps.opentelemetryApi
  api deps.opentelemetryLogging
```

---

Yeah that discussion wasn't about starter :) I think it should have out-of-the-box deps. Could even add otlp by default, reconfigure would be similar to the process for reconfiguring logging to use log4j instead of logback (exclude and add dependency)

---

> Could even add otlp by default, reconfigure would be similar to the process for reconfiguring logging to use log4j instead of logback (exclude and add dependency)

I like this idea. I opened #923 to track/discuss.